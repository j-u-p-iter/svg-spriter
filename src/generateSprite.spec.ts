import * as fs from "fs";
import * as path from "path";
import * as rimraf from "rimraf";
import { generateSprite, SpriteConfig } from ".";

describe("generateSprite", () => {
  let spriteConfig: SpriteConfig;
  let fileToCreatePath: string;

  beforeAll(() => {
    const pathToMDIcons = (basePath: string): string =>
      path.join(
        path.dirname(require.resolve("material-design-icons")),
        basePath
      );

    fileToCreatePath = path.join(__dirname, "./output/sprite.svg");

    spriteConfig = {
      category1: {
        success: pathToMDIcons(
          "/action/svg/production/ic_check_circle_24px.svg"
        ),
        compare: pathToMDIcons(
          "/action/svg/production/ic_compare_arrows_24px.svg"
        )
      },
      category2: {
        success: pathToMDIcons(
          "/action/svg/production/ic_check_circle_24px.svg"
        ),
        compare: pathToMDIcons(
          "/action/svg/production/ic_compare_arrows_24px.svg"
        )
      }
    };
  });

  afterEach(() => {
    const spriteDirPath = path.dirname(fileToCreatePath);

    if (fs.existsSync(spriteDirPath)) {
      rimraf.sync(path.dirname(fileToCreatePath));
    }
  });

  it("generates sprite properly", async () => {
    await generateSprite(
      { mode: { symbol: true } },
      spriteConfig,
      path.resolve(__dirname, fileToCreatePath)
    );

    const fileContent = fs.readFileSync(fileToCreatePath, {
      encoding: "utf-8"
    });

    expect(fileContent).toMatchSnapshot();
  });

  it("outputs error in case of error happens during sprite compiling", async () => {
    const consoleErrorSpy = jest
      .spyOn(console, "error")
      .mockImplementation(() => {});

    await generateSprite(
      { mode: { symbol: true } },
      spriteConfig,
      path.resolve(__dirname, "")
    );

    expect(consoleErrorSpy).toHaveBeenCalledTimes(1);

    consoleErrorSpy.mockRestore();
  });

  it("doesn`t throw when sprite config is empty", async () => {
    await generateSprite(
      { mode: { symbol: true } },
      {},
      path.resolve(__dirname, fileToCreatePath)
    );

    const fileContent = fs.readFileSync(fileToCreatePath, {
      encoding: "utf-8"
    });

    expect(fileContent).toMatchSnapshot();
  });
});
