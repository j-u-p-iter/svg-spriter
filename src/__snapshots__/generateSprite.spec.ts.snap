// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSprite doesn\`t throw when sprite config is empty 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"></svg>"`;

exports[`generateSprite generates sprite properly 1`] = `"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><symbol viewBox=\\"0 0 24 24\\" id=\\"category1-compare\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z\\"/></symbol><symbol viewBox=\\"0 0 24 24\\" id=\\"category1-success\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"/></symbol><symbol viewBox=\\"0 0 24 24\\" id=\\"category2-compare\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z\\"/></symbol><symbol viewBox=\\"0 0 24 24\\" id=\\"category2-success\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"/></symbol></svg>"`;
